google/cloud/storage/options.h

```diff
@@ -28,6 +28,28 @@
 
 namespace google {
 namespace cloud {
+namespace storage_experimental {
+inline namespace STORAGE_CLIENT_NS {
+/**
+ * Set the HTTP version used by the client.
+ *
+ * If this option is not provided, or is set to `default` then the library uses
+ * [libcurl's default], typically HTTP/2 with SSL. Possible settings include:
+ * - "1.0": use HTTP/1.0, this is not recommended as would require a new
+ *   connection for each request.
+ * - "1.1": use HTTP/1.1, this may be useful when the overhead of HTTP/2 is
+ *   unacceptable. Note that this may require additional connections.
+ * - "2TLS": use HTTP/2 with TLS
+ * - "2.0": use HTTP/2 with our without TLS.
+ *
+ * [libcurl's default]: https://curl.se/libcurl/c/CURLOPT_HTTP_VERSION.html
+ */
+struct CurlHttpVersion {
+  using Type = std::string;
```

Comment:
(optional) Did you consider making this option value a `long` and requiring that it's value be one of the CURL codes, like `CURL_HTTP_VERSION_2_0`? That may require that the caller include some curl header to access that symbol, which could be a bit annoying, but it would avoid ambiguity in the string format (like "2" works but "1" doesn't, but "1.0" and "2.0" both work).

Reply:
No, the thought did not cross my mind, thanks for bringing it up. On balance I think I will keep the string.  This is in `storage_experimental` partly because exposing the underlying implementation (libcurl) seems wrong.  If (big "if" I know) we used a different HTTP library: should we expose the values in that library?  If we use gRPC, what is the value of this option?  Using the CURL enums would be even worse (I think).

I do not have answers to those questions, so string and experimental keep things more loosely coupled until I do.

Reply:
Keeping string SGTM. I only suggested it because "Curl" is already in the option name, so the implementation is already exposed.

Reply:
I was going to suggest `int major; int minor` and taking "Curl" out of the option name, but I came back to preferring `string`.  Maybe "Curl" could still go in any case.

Reply:
Doh! Yeah, that `Curl` prefix was no good. Removed.